matlab/scripts/compute_pupil_periods.m

```diff
@@ -0,0 +1,62 @@
+%% choose an arbitrary pupil trace
+key.animal_id=8804;
+key.session=1;
+key.scan_idx=3;
+
+%% fetch timestamps, fetch and filter pupil radius trace 
+r = fetchn(preprocess.EyeTrackingFrame & key,'major_r');
+pupilTime = fetch1(preprocess.Eye & key,'eye_time');
+
+% interpolate over undetected frames
+rInterp = interp1(pupilTime(~isnan(r)),r(~isnan(r)),pupilTime);
+
+% low-pass filter below 1Hz
```

Comment:
You can probably replace this with a one-line standard way to low-pass filter below 1 Hz.  We use the mirror reflection of boundaries and fftfilt to meet specific performance goals in calcium signals.